notifications/broadcast.py

```diff
@@ -140,19 +133,150 @@ def validate_list_types(self, items):
+        print('not yet implemented')
+
+    return subject, message_text
+
+def ssl_cert_display_fields(cert_instance, noti_instance):
+    """
+    ssl_cert_display_fields message
+    """
+    noti_rule_msg = json.loads(noti_instance.rule_msg)
+    relationship = str(noti_rule_msg["relationship"])
+    subject_type = \
+    relationship.replace(" ", "").replace("\r", "").replace("\n", "").replace("\t", "").lower()
+
+    rule_action = {
+        "notyetvali": not_yet_valid(cert_instance, noti_rule_msg),
+        "willexpire": will_expire_in_less_than(cert_instance, noti_rule_msg),
+        "hasexpired": has_expired(cert_instance, noti_rule_msg)
+    }
+    subject, message = rule_action.get(subject_type[0:10],
+                                       "not yet implemented")
+    return subject, message
+
+def will_expire_in_less_than(cert_instance, noti_rule_msg):
+    """
+    This relationship build message for will exopir...rule
+    """
+    subject_list = []
+
+    days = str(noti_rule_msg["grace_period"]["days"])
+    relationship = str(noti_rule_msg["relationship"])
+
+    subject_list.append("Alert - %s" % (relationship))
+    subject_list.append("\nAn SSL Cert on %s " % \
+                       (str(cert_instance.common_name)))
+    subject_list.append(" port 443 will Expire in %s days" % (days))
+    
+    subject = " ".join(map(str, subject_list))
+    message = generate_ssl_cert_message(cert_instance,
+                                        noti_rule_msg)
+    return  subject, message
+
+def not_yet_valid(cert_instance, noti_rule_msg):
+    """
+    This relationship build message for not_yet_valid...rule
+    """
+    subject_list = []
+
+    relationship = str(noti_rule_msg["relationship"])
+
+    subject_list.append("Alert - %s" % (relationship))
+    subject_list.append("\nAn SSL Cert on %s " % \
+                       (str(cert_instance.common_name)))
+    subject_list.append(" port 443 is not valid anymore")
+    
+    subject = " ".join(map(str, subject_list))
+    message = generate_ssl_cert_message(cert_instance,
+                                        noti_rule_msg)
+    return  subject, message
+
+def has_expired(cert_instance, noti_rule_msg):
+    """
+    This relationship build message for has_expired...rule
+    """
+
+    subject_list = []
+
+    not_after = noti_rule_msg["facts"][1]
+    relationship = str(noti_rule_msg["relationship"])
+
+    subject_list.append("Alert - %s" % (relationship))
+    subject_list.append("\tAn SSL Cert on %s " % \
+                       (str(cert_instance.common_name)))
+    subject_list.append(" port 443 has already expired %s on" % not_after)
+    
+    subject = " ".join(map(str, subject_list))
+    message = generate_ssl_cert_message(cert_instance,
+                                        noti_rule_msg)
+    return  subject, message
+
+def generate_ssl_cert_message(cert_instance, noti_rule_msg):
+    """
+    Generate message for ssl-cert email
+    """
+
+    msg_list = []
```

Comment:
please changes this from using lists. it is an uncommon pattern for building large blocks of tests

```python
msg = initial_message
msg += '\n'
msg += next_chunk
# etc
```

or
```python
formatter = '{chunk_a}s some filler.\n{chunk_b}s some more filler'
msg = formatter % {'chunk_a': chunk_a, 'chunk_b': chunk_b}
```

or the equivalent with ''.format()